### PR TITLE
AzureRT updates

### PR DESCRIPTION
**Pull request by hyonholee**
Fix Set-AzureVMCustomExtension cmdlets
Fix VirtualMachineExtensionUsingHashTable test
Deprecate Name parameter of New-AzureVM cmdlets
Deprecate Get-AzureVMImageDetail and Get-AzureVMExtensionImageDetail cmdlets

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@hyonholee__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (hylee). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManager/Compute/Commands.Compute.Test/Commands.Compute.Test.csproj [hunk 1]
```diff
@@ -251,9 +251,13 @@
     <None Include="SessionRecords\Microsoft.Azure.Commands.Compute.Test.ScenarioTests.VirtualMachineExtensionTests\TestVirtualMachineCustomScriptExtension.json">
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </None>
+    <None Include="SessionRecords\Microsoft.Azure.Commands.Compute.Test.ScenarioTests.VirtualMachineExtensionTests\TestVirtualMachineCustomScriptExtensionFileUri.json">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </None>
     <None Include="SessionRecords\Microsoft.Azure.Commands.Compute.Test.ScenarioTests.VirtualMachineExtensionTests\TestVirtualMachineExtension.json">
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </None>
+    <None Include="SessionRecords\Microsoft.Azure.Commands.Compute.Test.ScenarioTests.VirtualMachineExtensionTests\TestVirtualMachineExtensionUsingHashTable.json" />
```
Comment (huangpf) [inline src/ResourceManager/Compute/Commands.Compute.Test/Commands.Compute.Test.csproj]: ```<CopyToOutputDirectory>Always</CopyToOutputDirectory>``` Needed.

**Comment by huangpf** [comment]
LGTM.

**Comment by amarzavery** [comment]
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/80/

**Comment by amarzavery** [comment]
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-msi/410/

**Pull request merged**